modules/git/repo_tag.go

```diff
@@ -34,7 +34,7 @@ func (repo *Repository) CreateTag(name, revision string) error {
 
 // CreateAnnotatedTag create one annotated tag in the repository
 func (repo *Repository) CreateAnnotatedTag(name, message, revision string) error {
-	_, err := NewCommand("tag", "-a", "-m", message, "--", name, revision).RunInDir(repo.Path)
+	_, err := NewCommand("tag", "-a", name, revision, "-m", message).RunInDir(repo.Path)
```

Comment:
Why the change in the order of arguments here?

From the git manual 

```
NAME
       git-tag - Create, list, delete or verify a tag object signed with GPG

SYNOPSIS
       git tag [-a | -s | -u <keyid>] [-f] [-m <msg> | -F <file>] [-e]
               <tagname> [<commit> | <object>]
       git tag -d <tagname>...
       git tag [-n[<num>]] -l [--contains <commit>] [--no-contains <commit>]
               [--points-at <object>] [--column[=<options>] | --no-column]
               [--create-reflog] [--sort=<key>] [--format=<format>]
               [--[no-]merged [<commit>]] [<pattern>...]
       git tag -v [--format=<format>] <tagname>...
```

You need to rearrange these arguments:

```suggestion
	_, err := NewCommand("tag", "-a", "-m", message, name, revision).RunInDir(repo.Path)
```

However, there really should be a way of terminating the options, usually that is through `--`. Does that really not work here? if it doesn't we need to ensure that both name and revision do not start with an initial `-`.

Reply:
Otherwise you can cause a hang simply by passing in a name or revision equal to `-e`

Reply:
Sorry, I forget it, It's not an necessary change. I have change it to previous version now.